app/controllers/collections_controller.rb

```diff
@@ -504,11 +525,18 @@ def update_document_source_in_graph(jsonld_graph, doc_identifier, doc_source)
     jsonld_graph
   end
 
+  #
+  # Returns a hash containing the metadata for a document source
+  #
+  def format_document_source_metadata(doc_source)
+    # Escape any filename spaces with '%20' as URIs with spaces are flagged as invalid when RDF loads
+    {'@id' => "file://#{doc_source.sub(" ", "%20")}"}
```

Comment:
Okay for now, but I think that Rails has it's own escape methods so you don't need to reinvent the wheel. There are probably other cases that also need to be escaped, such as having an ampersand in the file name.